master/internal/db/postgres_experiments.go

```diff
@@ -876,14 +876,30 @@ func (db *PgDB) DeleteExperiments(ctx context.Context, ids []int) error {
 		return nil
 	}
 
-	var deletedIDs []int32
-	_, err := Bun().NewDelete().Model(&deletedIDs).Table("experiments").
-		Where("id IN (?)", bun.In(ids)).
-		Returning("id").
-		Exec(ctx)
+	err := Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		if _, err := tx.NewDelete().Model(&model.CheckpointV2{}).
+			Where(`task_id IN (
+	SELECT tt.task_id
+	FROM run_id_task_id tt
+	JOIN trials t ON t.id = tt.run_id
+	WHERE experiment_id IN (?)
+)`, bun.In(ids)).
+			Exec(ctx); err != nil {
```

Comment:
I wish we had some defensive programming mechanism which ensures that the checkpoint data is deleted from storage when a checkpoint is deleted from the db. otherwise it's not super obvious this code is correct

Reply:
could we add a trigger that aborts sql checkpoint delete if the state isn't deleted?

Reply:
Created https://hpe-aiatscale.atlassian.net/browse/DET-10027